File: leetcode/3.无重复字符的最长子串.go
func lengthOfLongestSubstring(s string) int {
    l, r := 0, 0
    mp := map[byte]int{}
    ans := 0
    for r < len(s) {
        mp[s[r]] += 1
        for mp[s[r]] > 1 {
            mp[s[l]] -= 1
            l += 1
        } 
        ans = max(ans, r - l + 1)
        r += 1
    }
    return ans
}

File: leetcode/3.无重复字符的最长子串.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        n=len(s)
        st=set()
        i,j=0,0
        ans=0
        while i<n and j<n:
            if i!=0:
                st.remove(s[i-1])
            while j<n and s[j] not in st:
                st.add(s[j])
                j+=1
            ans=max(ans,j-i)
            i+=1
        return ans